### DISCUSSION TITLE
UX bugs & improvements

### DISCUSSION THREAD

**Discussion by pun-ky** [websight-io/starter#11] [Ideas]
**Describe the bug**

1. "Administration" page has no layout (top bar, left panel, etc)

![image](https://user-images.githubusercontent.com/10673561/194149718-1555f9d8-20e1-4e99-9f74-f084089c4785.png)

2. There is no "Alt" key on Mac / Groovy Console

![image](https://user-images.githubusercontent.com/10673561/194149827-55f63c23-6bb2-41b0-84d7-f8407f360eb1.png)

and usually the "Command/Ctrl + Enter" is used in similar tools to trigger execution

3. Clicking system version does not list the system components

![image](https://user-images.githubusercontent.com/10673561/194150029-42aa27a6-d742-40fe-b606-180fd890ab3d.png)

4. Where is a link to "/system/console" ?:) 

5. Pretty hard to edit the page

![image](https://user-images.githubusercontent.com/10673561/194151391-e5ce34fa-8bf9-4358-8ec4-dc746605708f.png)

2 clicks are needed to check the checkbox then click "edit" on top bar
or
move cursor totally to the right and click pencil icon; I was suspecting that clicking on the page name will open edit mode; I suggest to maybe provide an option to not be forced to move mouse each time to the right but move this dropdown menu somehow to the left which will eliminate the need from jumping continuosly cursor from left (when navigating between pages) to the right (when wanting to open edit mode / most often operation)

right now it's pretty frustrating (exactly in the same way as on AEM) ;) 

what is more, according to my screen resolution, I have enough space to display more action icons than this pencil alone which looks a little bit strange IMO 

6. Footer is not sticky (to the bottom)

![image](https://user-images.githubusercontent.com/10673561/194152466-76bb598a-55aa-4e82-b99c-3ac797ef2c4f.png)

7. Some dropdowns have icons, some other not
 
![image](https://user-images.githubusercontent.com/10673561/194152666-20251e42-dbf2-468e-8a41-f1ec6a4ee24d.png)

![image](https://user-images.githubusercontent.com/10673561/194152715-4f5da47b-cefa-4fc7-b19a-86906c7b95b0.png)

8. Assets browser cuts off image thumbs horizontally

![image](https://user-images.githubusercontent.com/10673561/194153376-07ae0133-154f-4d8a-8389-a74c34bdc814.png)

9. Double-click does not open edit dialog

This is super important! Really! ;)

![image](https://user-images.githubusercontent.com/10673561/194152989-beb42481-befb-4062-be1b-754109d04c48.png)

10. Image url choose path - no idea how it works

![image](https://user-images.githubusercontent.com/10673561/194153715-559920fe-b363-482a-a7bb-17dbdae5899e.png)

Maybe some picker should be opened somehow but I have no clue how use it

11. How to add a new accordion item?

![image](https://user-images.githubusercontent.com/10673561/194154168-ab6698d2-6d0f-44c1-abaa-ec715d925faf.png)

I have no clue...

edit: now I have; d&d from "Components" panel ... but in WP Gutenberg there is plus icon; on AEM is "Add components here" or sth like that; without such "helpers" it is not so intuitive to demystify it

12. Luna bestsellers too wide

![image](https://user-images.githubusercontent.com/10673561/194154353-428203ed-ccbd-4e67-bf9d-4ee1dc3a812e.png)

13. Clicking component in the tree should scroll the window to it

![image](https://user-images.githubusercontent.com/10673561/194155004-a47f4f99-9a9f-45bb-b92b-fc01a89167d4.png)

now it only highlights it; if in the viewport it is not visible it is needed to scroll it on our own unnecessarily

14. View published link/button

![image](https://user-images.githubusercontent.com/10673561/194155399-58f514b1-fd9d-4379-8dd4-1620701120fd.png)

to be able to quickly jump from

http://localhost:8080/apps/websight/index.html/content/luna/pages/Homepage::editor to http://localhost:8080/content/luna/pages/Homepage.html

15. Download SVG has no effect

nothing happening

(btw browser should be capable to display SVG... why Websight Assets is not trying to display it ?:))

![image](https://user-images.githubusercontent.com/10673561/194156833-9b11a5b7-45ad-49cc-bd97-28dd437b0828.png)

that download button does not work too

![image](https://user-images.githubusercontent.com/10673561/194157400-e5db04eb-ee92-4698-97fd-3729b917cb19.png)

16. Asset preview dropdown is cut off and has no icons

![image](https://user-images.githubusercontent.com/10673561/194157134-c9ab3be8-b684-4b77-a4a0-bb07c6d9b59b.png)

probably `right` class is missing :) 

17. Uploading file

a) after uploading a file with unique name / successfully - the asset on the list should be highlighted or sth
b) rename file dialog should have "Rename" button disabled to indicate that the name should be changed, also the dialog could auto-suggest the new name like OS/Windows/Mac does by appending "-1" or sth

![image](https://user-images.githubusercontent.com/10673561/194158224-c0fe7d62-fc4a-4848-934e-5374ec847d92.png)

**WebSight CMS version**

first publicly available community version; not able to retrieve that info from GUI

![image](https://user-images.githubusercontent.com/10673561/194150416-6c1ce27b-7b21-43a8-9069-b92780980ef7.png)

**Desktop (please complete the following information):**
 - OS: Mac
 - Browser: Chrome
 - Version: most up-to-date

====

General feedback,

UX looks very nice. There are only a few glitches / inconsistencies but probably relatively easy to fix.
I am impressed how quickly works page editing. Good job ;) AEM is a turtle here...

**Comment by kamil-kwarciak** [comment]
Hi @pun-ky. Thank you for the review and the valuable feedback. We have some of the pointed issues in our backlog. Others are new, and we will include them during further development. As we have plenty of items reported here, I will convert this issue into a discussion. Later, I will create a separate ticket for each topic to track them easily. Once again, thank you for your contribution. It is very appreciated.

**Reply by michalcukierman** [reply]
I put some comments on the tickets. Nice catches.
